lib/bundler_api/update/gem_db_helper.rb

```diff
@@ -62,7 +62,9 @@ def find_or_insert_version(spec, rubygem_id, platform = 'ruby', indexed = nil)
     version    = @db[:versions].filter(
       rubygem_id: rubygem_id,
       number:     spec.version.version,
-      platform:   platform
+      platform:   platform,
+      rubygems_version: spec.required_rubygems_version.none? ? nil : spec.required_rubygems_version.to_s,
+      required_ruby_version: spec.required_ruby_version.none? ? nil : spec.required_ruby_version.to_s
```

Comment:
I think we also want to get the checksum from RubyGems.org, but we might have to make an API call for it. :/